### PR TITLE
Support providing MLB templates via environment variables rather than…

### PR DESCRIPTION
**Pull request by justinrlee**
… template tgz files.

https://jira.mesosphere.com/browse/DCOS-18367

### CODE REVIEW

**Comment by nlsun** [comment]
It's rather difficult to squeeze text into environment variables because of the kind of escaping is required to get it to fit into JSON (e.g. `\\n`), so not terribly comfortable with this change. But at the same time it isn't exactly detrimental, I suppose.

**Comment by justinrlee** [comment]
Yeah so I think I handled it so that it will accept either `\n` or `\\n` (or even `\\\n`, not sure on that one).  This is primarily to overwrite changes that only require a line or two.  But I've tested it as far as overwriting the HAPROXY_HEAD thing with no issues.

File: config.py [hunk 1]
```diff
@@ -770,17 +770,25 @@ def __init__(self, directory='templates'):
...
-                filename = os.path.join(self.__template_directory, name)
-                with open(filename) as f:
-                    logger.info('overriding %s from %s', name, filename)
-                    self.t[template].value = f.read()
-            except IOError:
-                logger.debug("setting default value for %s", name)
+            if os.environ.get(name):
+                logger.info('overriding %s from environment variable', name)
+                env_template_val = os.environ.get(name)
+                # logger.info(env_template_val)
```
Comment (drewkerrigan) [inline config.py]: Looks like a leftover debug line? Please remove if so

File: config.py [hunk 2]
```diff
@@ -770,17 +770,25 @@ def __init__(self, directory='templates'):
...
         for template in self.t:
             name = self.t[template].full_name
-            try:
-                filename = os.path.join(self.__template_directory, name)
-                with open(filename) as f:
-                    logger.info('overriding %s from %s', name, filename)
-                    self.t[template].value = f.read()
-            except IOError:
-                logger.debug("setting default value for %s", name)
+            if os.environ.get(name):
```
Comment (drewkerrigan) [inline config.py]: Would it be possible to add a test to `tests/test_marathon_lb.py`? You should be able to do something like

```
os.environ["key"] = "val"
```

**Comment by justinrlee** [comment]
@drewkerrigan I've removed the debug line and added a basic unit test - can you validate?

Thanks!

**Pull request merged**